tagmanager/ctags/powershell.c

```diff
@@ -0,0 +1,658 @@
+				fileUngetc (next);
+			else
+				c = next;
+		}
+	} while (c != EOF && c != '\n' && c != '\r');
+	return c;
+}
+
+static void readToken (tokenInfo *const token)
+{
+	int c;
+
+	token->type		= TOKEN_UNDEFINED;
+	vStringClear (token->string);
+
+getNextChar:
+
+	c = fileGetc ();
+	c = skipWhitespaces (c);
+
+	token->lineNumber   = getSourceLineNumber ();
+	token->filePosition = getInputFilePosition ();
+
+	switch (c)
+	{
+		case EOF: token->type = TOKEN_EOF;					break;
+		case '(': token->type = TOKEN_OPEN_PAREN;			break;
+		case ')': token->type = TOKEN_CLOSE_PAREN;			break;
+		case ';': token->type = TOKEN_SEMICOLON;			break;
+		case ',': token->type = TOKEN_COMMA;				break;
+		case '.': token->type = TOKEN_PERIOD;				break;
+		case ':': token->type = TOKEN_COLON;				break;
+		case '{': token->type = TOKEN_OPEN_CURLY;			break;
+		case '}': token->type = TOKEN_CLOSE_CURLY;			break;
+		case '[': token->type = TOKEN_OPEN_SQUARE;			break;
+		case ']': token->type = TOKEN_CLOSE_SQUARE;			break;
+
+		case '=':
+		{
+			int d = fileGetc ();
+			if (d == '=' || d == '>')
+				token->type = TOKEN_OPERATOR;
+			else
+			{
+				fileUngetc (d);
+				token->type = TOKEN_EQUAL_SIGN;
+			}
+			break;
+		}
+
+		case '\'':
+		case '"':
+			token->type = TOKEN_STRING;
+			parseString (token->string, c);
+			token->lineNumber = getSourceLineNumber ();
+			token->filePosition = getInputFilePosition ();
+			break;
+
+		case '<':
+		{
+			int d = fileGetc ();
+			if (d == '#')
+			{
+				/* <# ... #> multiline comment */
+				do
+				{
+					c = skipToCharacter ('#');
+					if (c != EOF)
+					{
+						c = fileGetc ();
+						if (c == '>')
+							break;
+						else
+							fileUngetc (c);
+					}
+				} while (c != EOF && c != '\0');
+				goto getNextChar;
+			}
+			else
+			{
+				fileUngetc (d);
+				token->type = TOKEN_UNDEFINED;
+			}
+			break;
+		}
+
+		case '#': /* comment */
+			skipSingleComment ();
+			goto getNextChar;
+			break;
+
+		case '+':
+		case '-':
+		case '*':
+		case '/':
+		case '%':
+		{
+			int d = fileGetc ();
+			if (d != '=')
+				fileUngetc (d);
+			token->type = TOKEN_OPERATOR;
+			break;
+		}
+
+		case '$': /* variable start */
+		{
+			int d = fileGetc ();
+			if (! isIdentChar (d))
+			{
+				fileUngetc (d);
+				token->type = TOKEN_UNDEFINED;
+			}
+			else
+			{
+				parseIdentifier (token->string, d);
+				token->type = TOKEN_VARIABLE;
+			}
+			break;
+		}
+
+		default:
+			if (! isIdentChar (c))
+				token->type = TOKEN_UNDEFINED;
+			else
+			{
+				parseIdentifier (token->string, c);
+				if (isTokenFunction (token->string))
+					token->type = TOKEN_KEYWORD;
+				else
+					token->type = TOKEN_IDENTIFIER;
+			}
+			break;
+	}
+}
+
+static void enterScope (tokenInfo *const parentToken,
+						const vString *const extraScope,
+						const int parentKind);
+
+/* strip a possible PowerShell scope specification and convert it to accessType */
+static const char *const parsePowerShellScope (tokenInfo *const token)
```

Comment:
`$Local : ALocalVar = $null`
gives: `Unexpected token ':' in expression or statement.`
So I guess whitespaces are not allowed in PowerShell.

Considering the colon as an identifier character just made the implementation simpler.
While testing this, I noticed something like `$Global:variable:name` seems valid syntax in PowerShell and Geany would parse it as 'variable:name' in the symbol tree. However, this is either a bug in PowerShell or is has a special meaning I don't know. Because 'write-host $variable:name' doesn't print anything, so either this is just invalid or the second colon does something special.
I'd like to keep the code here as it is, at least until we know better what the syntax with two colons in a variable defintion actually means. I'll ask the Windows Admin at my work, maybe he has some clue.

While testing, I also noticed the part before thr first colon can not only be a scope like Gloabl, Local or Private, but also a PSDrive (e.g. a drive letter). Scary PowerShell, I'd say. Though, I think this is not relevant for the tag parsing here.